File: src/main/java/ru/tanaxxt/currencysystem/entities/Currency.java
package ru.tanaxxt.currencysystem.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Entity
@Getter
@Setter
public class Currency {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String baseCurrency;

    @Column(nullable = false)
    private String priceChangeRange;

    @Column()
    private String description;

    @Column()
    private boolean isDeleted;
}

File: src/main/java/ru/tanaxxt/currencysystem/entities/Currency.kt
package ru.tanaxxt.currencysystem.entities

import jakarta.persistence.*
import java.util.UUID

@Entity
data class Currency(
    @Column(nullable = false)
    var name: String = "",

    @Column(nullable = false, unique = true)
    var baseCurrency: String = "",

    @Column(nullable = false)
    var priceChangeRange: String = "",

    @Column
    var description: String? = null,
) {
    @Id
    var id: UUID = UUID.randomUUID()

    @Column
    var isDeleted: Boolean = false
}
